plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -1,23 +1,145 @@
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdkAppEngineDevServer;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessExitListener;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessStartListener;
+
+/**
+ * A {@link ServerBehaviourDelegate} for App Engine Server executed via the Java App Management
+ * Client Library.
+ */
 public class LocalAppEngineServerBehaviour extends ServerBehaviourDelegate {
+  private LocalAppEngineStartListener localAppEngineStartListener;
+  private LocalAppEngineExitListener localAppEngineExitListener;
+  private AppEngineDevServer devServer;
+
+  public LocalAppEngineServerBehaviour () {
+	localAppEngineStartListener = new LocalAppEngineStartListener();
+	localAppEngineExitListener = new LocalAppEngineExitListener();
+  }
 
   @Override
   public void stop(boolean force) {
     setServerState(IServer.STATE_STOPPING);
-    // TODO stop server
+    // stop server
+    terminate();
     setServerState(IServer.STATE_STOPPED);    
   }
 
-  void setStarted() {
-    setServerState(IServer.STATE_STARTED);    
+  /**
+   * Convenience method allowing access to protected method in superclass.
+   */
+  @Override
+  protected IModuleResourceDelta[] getPublishedResourceDelta(IModule[] module) {
+    return super.getPublishedResourceDelta(module);
   }
-  
-  void setStarting() {
-    setServerState(IServer.STATE_STARTING);    
+
+  /**
+   * Convenience method allowing access to protected method in superclass.
+   */
+  @Override
+  protected IModuleResource[] getResources(IModule[] module) {
+    return super.getResources(module);
+  }
+
+  /**
+   * Returns runtime base directory. Uses temp directory.
+   */
+  public IPath getRuntimeBaseDirectory() {
+    return getTempDirectory(false);
+  }
+
+  /**
+   * @return the directory at which module will be published.
+   */
+  public IPath getModuleDeployDirectory(IModule module) {
+    return getRuntimeBaseDirectory().append(module.getName());
+  }
+
+  /**
+   * Convenience accessor to protected member in superclass.
+   */
+  public void setModulePublishState2(IModule[] module, int state) {
+	  setModulePublishState(module, state);
+  }
+
+  void startDevServer(List<File> runnables, MessageConsoleStream stream) {
+	  setServerState(IServer.STATE_STARTING);
+
+	  LocalAppEngineOutputLineListener outputListener =
+			  new LocalAppEngineOutputLineListener(stream);
+
+      // Create dev app server instance
+	  CloudSdk cloudSdk = new CloudSdk.Builder()
+			  .addStdOutLineListener(outputListener)
+			  .addStdErrLineListener(outputListener)
+			  .startListener(localAppEngineStartListener)
+			  .exitListener(localAppEngineExitListener)
+			  .build();
+	  devServer = new CloudSdkAppEngineDevServer(cloudSdk);
+	  DefaultRunConfiguration devServerRunConfiguration = new DefaultRunConfiguration();
+	  devServerRunConfiguration.setAppYamls(runnables);
```

Comment:
Created issue for this: https://github.com/GoogleCloudPlatform/app-tools-lib-for-java/issues/132